workers/unity/Packages/com.improbable.gdk.core/Commands/WorldCommands/CreateEntity.cs

```diff
@@ -441,6 +444,17 @@ public List<ReceivedResponse> GetResponsesReceivedForEntity(Unity.Entities.Entit
                     return false;
                 }
 
+                public void RegisterResponseTask(long requestId, TaskCompletionSource<ReceivedResponse> task)
```

Comment:
Not very familiar with C#/Unity threading model - are `RegisterResponseTask` and `AddResponse` always called from the same thread?
If not, is there a chance of race condition when we try to mutate `internalRequestIdToReceivedResponseTask` dictionary from different threads at the same time?